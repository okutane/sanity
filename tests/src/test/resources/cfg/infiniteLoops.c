void labelBeforeGoTo() {
start:
	goto start;
}

void conditionalBreak() {
	for (int i = 0; ; ++i) {
		if (i >= 10) {
			break;
		}
	}
}
