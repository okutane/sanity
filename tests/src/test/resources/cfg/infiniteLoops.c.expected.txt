CFG: labelBeforeGoTo
#0000:	 noop next: #0000:
 - no source -

CFG: conditionalBreak
#0000:	 alloc: i
infiniteLoops.c:7 for (int i = 0; ; ++i) {
#0001:	 assign: *i = 0
infiniteLoops.c:7 for (int i = 0; ; ++i) {
#0002:	 noop
 - no source -
#0003:	 assign: tmp0 = *i >= 10
infiniteLoops.c:8 if (i >= 10) {
#0004:	 if: tmp0 then #0005:	 else #0006:
infiniteLoops.c:8 if (i >= 10) {
#0005:	 return <exit>
infiniteLoops.c:12 }
#0006:	 assign: tmp1 = *i + 1
infiniteLoops.c:7 for (int i = 0; ; ++i) {
#0007:	 assign: *i = tmp1 next: #0002:
infiniteLoops.c:7 for (int i = 0; ; ++i) {

